File: spring-skeleton/frontend/views/vista-logueado.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/horizontal-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/button/src/vaadin-button.js';
import '@vaadin/vertical-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';

@customElement('vista-logueado')
export class VistaLogueado extends LitElement {

  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-horizontal-layout class="content" style="width: 100%; height: 100%;position:absolute" id="vaadinHorizontalLayout">
 <vaadin-horizontal-layout class="content" style="width: 100%; height: 100%;" id="vaadinHorizontalLayout1">
  <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; width: 30%;" id="verticalLayout-izquierdo">
   <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; margin-left: var(--lumo-space-m); width: 100%;" id="horizontalLayout-ListaUsuariosMasFamosos">
    <label style="height: 20%; width: 100%; flex-grow: 1; margin: var(--lumo-space-xs); font-size: 15px;" id="label-ListaUsuariosMasFamosos">Lista de usuarios más famosos</label>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="flex-grow: 1; height: 75%; flex-shrink: 1; width: 70%; align-self: center; flex-direction: column; align-items: stretch;" id="horizontalLayout-ListaUsuariosMasFamosos2"></vaadin-horizontal-layout>
   <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1;" id="horizontalLayout-VerMasUsuarios">
    <vaadin-button style="flex-grow: 1; padding: var(--lumo-space-m); margin-left: var(--lumo-space-m); margin-bottom: var(--lumo-space-s); align-self: flex-start;" tabindex="0" id="button-VerMasUsuarios">
      Ver más usuarios 
    </vaadin-button>
   </vaadin-horizontal-layout>
  </vaadin-vertical-layout>
  <vaadin-vertical-layout style="flex-grow: 1; align-items: center; justify-content: space-around;" id="verticalLayout-central">
   <vaadin-horizontal-layout style="flex-grow: 1; height: 9%;" id="horizontalLayout-centralSuperior">
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; width: 15%; margin-top: 23px;" id="verticalLayout-CerrarSesionNotificaciones">
     <vaadin-button style="flex-grow: 0; max-width: none; align-self: flex-end; Height:30px;" tabindex="0" id="button-CerrarSesion">
       Cerrar Sesión 
     </vaadin-button>
     <vaadin-button style="flex-grow: 0; max-width: none; align-self: flex-end; Height:30px;" tabindex="0" id="button-Notificaciones">
       Notificaciones 
     </vaadin-button>
    </vaadin-vertical-layout>
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; align-self: flex-start;" id="verticalLayout-imagenPerfil">
     <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1;" id="horizontalLayout-ImagenPerfil">
      <img id="img-PortadaTwitter" style="flex-grow: 1; width: 25%; margin-top: var(--lumo-space-m); margin-left: var(--lumo-space-s); height: 50%; align-self: flex-start;">
     </vaadin-horizontal-layout>
    </vaadin-vertical-layout>
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; width: 15%; max-height: 25%;" id="verticalLayout-BaneadoVerperfil">
     <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; margin-top: var(--lumo-space-l); max-height: 50%;" id="horizontalLayout-Baneados">
      <vaadin-button tabindex="0" style="flex-grow: 1; align-self: flex-start; Height:30px; width: 200%;" id="button-Baneados">
        Baneados 
      </vaadin-button>
      <img id="img-PerfilUsuario" style="flex-grow: 1; width: 100%; height: 100%; margin-left: var(--lumo-space-s);">
     </vaadin-horizontal-layout>
     <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1;" id="horizontalLayout-VerPerfilEscribirTweet">
      <vaadin-vertical-layout theme="spacing" id="verticalLayout-VerPerfilEscribirTweet">
       <vaadin-button style="flex-grow: 1; Height:30px; align-self: flex-start;" tabindex="0" id="button-VerPerfil">
         Ver perfil 
       </vaadin-button>
       <vaadin-button style="flex-grow: 1; width: 100%; Height:30px; align-self: flex-start;" tabindex="0" id="button-EscribirTweet">
         Escribir tweet 
       </vaadin-button>
      </vaadin-vertical-layout>
     </vaadin-horizontal-layout>
    </vaadin-vertical-layout>
   </vaadin-horizontal-layout>
   <vaadin-vertical-layout style="flex-grow: 1; align-items: stretch; align-self: center; flex-direction: column; width: 100%; margin-top: var(--lumo-space-xl); padding-top: var(--lumo-space-xl);" id="verticalLayout-inferior"></vaadin-vertical-layout>
  </vaadin-vertical-layout>
  <vaadin-vertical-layout style="flex-grow: 1; width: 35%;" id="verticalLayout-derecho" theme="spacing">
   <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 100%;" id="horizontalLayout-ListaHashtagsUsados">
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; margin-right: var(--lumo-space-m);" id="verticalLayout-listaHashtagsMasUsados">
     <div style="align-self: flex-end; margin-right: var(--lumo-space-xs); margin: var(--lumo-space-xs); flex-grow: 1; font-size: 15px;" id="label-ListaDeHashtagsMasUsados">
       Lista de Hashtags más usados 
     </div>
    </vaadin-vertical-layout>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="flex-grow: 1; height: 75%; width: 70%; align-self: center; flex-direction: column; align-items: stretch;" id="horizontalLayout-HashtagsMasUsados"></vaadin-horizontal-layout>
   <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 100%;" id="horizontalLayout-VerMasHashtags">
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="verticaLayout-VerMásHashTags">
     <vaadin-button style="flex-grow: 0; padding: var(--lumo-space-m); align-self: flex-end; margin-bottom: var(--lumo-space-s); margin-right: var(--lumo-space-m); flex-shrink: 1;" tabindex="0" id="button-VerMasHashtags">
       Ver más hashtags 
     </vaadin-button>
    </vaadin-vertical-layout>
   </vaadin-horizontal-layout>
  </vaadin-vertical-layout>
 </vaadin-horizontal-layout>
</vaadin-horizontal-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: spring-skeleton/src/main/java/vistas/VistaLogueado.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.html.Div;

@Tag("vista-logueado")
@JsModule("./views/vista-logueado.ts")
public class VistaLogueado extends LitTemplate {

	
	@Id("vaadinHorizontalLayout")
	private HorizontalLayout vaadinHorizontalLayout;
	@Id("vaadinHorizontalLayout1")
	private HorizontalLayout vaadinHorizontalLayout1;
	@Id("verticalLayout-izquierdo")
	private Element verticalLayoutIzquierdo;
	@Id("horizontalLayout-ListaUsuariosMasFamosos")
	private HorizontalLayout horizontalLayoutListaUsuariosMasFamosos;
	@Id("label-ListaUsuariosMasFamosos")
	private Label labelListaUsuariosMasFamosos;
	@Id("horizontalLayout-ListaUsuariosMasFamosos2")
	private HorizontalLayout horizontalLayoutListaUsuariosMasFamosos2;
	@Id("horizontalLayout-VerMasUsuarios")
	private HorizontalLayout horizontalLayoutVerMasUsuarios;
	@Id("button-VerMasUsuarios")
	private Button buttonVerMasUsuarios;
	@Id("verticalLayout-central")
	private Element verticalLayoutCentral;
	@Id("horizontalLayout-centralSuperior")
	private HorizontalLayout horizontalLayoutCentralSuperior;
	@Id("verticalLayout-CerrarSesionNotificaciones")
	private Element verticalLayoutCerrarSesionNotificaciones;
	@Id("button-CerrarSesion")
	private Button buttonCerrarSesion;
	@Id("button-Notificaciones")
	private Button buttonNotificaciones;
	@Id("verticalLayout-imagenPerfil")
	private Element verticalLayoutImagenPerfil;
	@Id("horizontalLayout-ImagenPerfil")
	private HorizontalLayout horizontalLayoutImagenPerfil;
	@Id("img-PortadaTwitter")
	private Image imgPortadaTwitter;
	@Id("horizontalLayout-centralImagen")
	private HorizontalLayout horizontalLayoutCentralImagen;
	@Id("verticalLayout-BaneadoVerperfil")
	private Element verticalLayoutBaneadoVerperfil;
	@Id("horizontalLayout-Baneados")
	private HorizontalLayout horizontalLayoutBaneados;
	@Id("button-Baneados")
	private Button buttonBaneados;
	@Id("img-PerfilUsuario")
	private Image imgPerfilUsuario;
	@Id("horizontalLayout-VerPerfilEscribirTweet")
	private HorizontalLayout horizontalLayoutVerPerfilEscribirTweet;
	@Id("verticalLayout-VerPerfilEscribirTweet")
	private Element verticalLayoutVerPerfilEscribirTweet;
	@Id("button-VerPerfil")
	private Button buttonVerPerfil;
	@Id("button-EscribirTweet")
	private Button buttonEscribirTweet;
	@Id("horizontalLayout-inferior")
	private HorizontalLayout horizontalLayoutInferior;
	@Id("verticalLayout-inferior")
	private Element verticalLayoutInferior;
	@Id("verticalLayout-derecho")
	private Element verticalLayoutDerecho;
	@Id("horizontalLayout-ListaHashtagsUsados")
	private HorizontalLayout horizontalLayoutListaHashtagsUsados;
	@Id("verticalLayout-listaHashtagsMasUsados")
	private Element verticalLayoutListaHashtagsMasUsados;
	@Id("label-ListaDeHashtagsMasUsados")
	private Div labelListaDeHashtagsMasUsados;
	@Id("horizontalLayout-HashtagsMasUsados")
	private HorizontalLayout horizontalLayoutHashtagsMasUsados;
	@Id("horizontalLayout-VerMasHashtags")
	private HorizontalLayout horizontalLayoutVerMasHashtags;
	@Id("verticaLayout-VerMásHashTags")
	private Element verticaLayoutVerMásHashTags;
	@Id("button-VerMasHashtags")
	private Button buttonVerMasHashtags;

	public VistaLogueado() {
        // You can initialise any data required for the connected UI components here.
    }

	public HorizontalLayout getVaadinHorizontalLayout() {
		return vaadinHorizontalLayout;
	}

	public void setVaadinHorizontalLayout(HorizontalLayout vaadinHorizontalLayout) {
		this.vaadinHorizontalLayout = vaadinHorizontalLayout;
	}

	public HorizontalLayout getVaadinHorizontalLayout1() {
		return vaadinHorizontalLayout1;
	}

	public void setVaadinHorizontalLayout1(HorizontalLayout vaadinHorizontalLayout1) {
		this.vaadinHorizontalLayout1 = vaadinHorizontalLayout1;
	}

	public Element getVerticalLayoutIzquierdo() {
		return verticalLayoutIzquierdo;
	}

	public void setVerticalLayoutIzquierdo(Element verticalLayoutIzquierdo) {
		this.verticalLayoutIzquierdo = verticalLayoutIzquierdo;
	}

	public HorizontalLayout getHorizontalLayoutListaUsuariosMasFamosos() {
		return horizontalLayoutListaUsuariosMasFamosos;
	}

	public void setHorizontalLayoutListaUsuariosMasFamosos(HorizontalLayout horizontalLayoutListaUsuariosMasFamosos) {
		this.horizontalLayoutListaUsuariosMasFamosos = horizontalLayoutListaUsuariosMasFamosos;
	}

	public Label getLabelListaUsuariosMasFamosos() {
		return labelListaUsuariosMasFamosos;
	}

	public void setLabelListaUsuariosMasFamosos(Label labelListaUsuariosMasFamosos) {
		this.labelListaUsuariosMasFamosos = labelListaUsuariosMasFamosos;
	}

	public HorizontalLayout getHorizontalLayoutListaUsuariosMasFamosos2() {
		return horizontalLayoutListaUsuariosMasFamosos2;
	}

	public void setHorizontalLayoutListaUsuariosMasFamosos2(HorizontalLayout horizontalLayoutListaUsuariosMasFamosos2) {
		this.horizontalLayoutListaUsuariosMasFamosos2 = horizontalLayoutListaUsuariosMasFamosos2;
	}

	public HorizontalLayout getHorizontalLayoutVerMasUsuarios() {
		return horizontalLayoutVerMasUsuarios;
	}

	public void setHorizontalLayoutVerMasUsuarios(HorizontalLayout horizontalLayoutVerMasUsuarios) {
		this.horizontalLayoutVerMasUsuarios = horizontalLayoutVerMasUsuarios;
	}

	public Button getButtonVerMasUsuarios() {
		return buttonVerMasUsuarios;
	}

	public void setButtonVerMasUsuarios(Button buttonVerMasUsuarios) {
		this.buttonVerMasUsuarios = buttonVerMasUsuarios;
	}

	public Element getVerticalLayoutCentral() {
		return verticalLayoutCentral;
	}

	public void setVerticalLayoutCentral(Element verticalLayoutCentral) {
		this.verticalLayoutCentral = verticalLayoutCentral;
	}

	public HorizontalLayout getHorizontalLayoutCentralSuperior() {
		return horizontalLayoutCentralSuperior;
	}

	public void setHorizontalLayoutCentralSuperior(HorizontalLayout horizontalLayoutCentralSuperior) {
		this.horizontalLayoutCentralSuperior = horizontalLayoutCentralSuperior;
	}

	public Element getVerticalLayoutCerrarSesionNotificaciones() {
		return verticalLayoutCerrarSesionNotificaciones;
	}

	public void setVerticalLayoutCerrarSesionNotificaciones(Element verticalLayoutCerrarSesionNotificaciones) {
		this.verticalLayoutCerrarSesionNotificaciones = verticalLayoutCerrarSesionNotificaciones;
	}

	public Button getButtonCerrarSesion() {
		return buttonCerrarSesion;
	}

	public void setButtonCerrarSesion(Button buttonCerrarSesion) {
		this.buttonCerrarSesion = buttonCerrarSesion;
	}

	public Button getButtonNotificaciones() {
		return buttonNotificaciones;
	}

	public void setButtonNotificaciones(Button buttonNotificaciones) {
		this.buttonNotificaciones = buttonNotificaciones;
	}

	public Element getVerticalLayoutImagenPerfil() {
		return verticalLayoutImagenPerfil;
	}

	public void setVerticalLayoutImagenPerfil(Element verticalLayoutImagenPerfil) {
		this.verticalLayoutImagenPerfil = verticalLayoutImagenPerfil;
	}

	public HorizontalLayout getHorizontalLayoutImagenPerfil() {
		return horizontalLayoutImagenPerfil;
	}

	public void setHorizontalLayoutImagenPerfil(HorizontalLayout horizontalLayoutImagenPerfil) {
		this.horizontalLayoutImagenPerfil = horizontalLayoutImagenPerfil;
	}

	public Image getImgPortadaTwitter() {
		return imgPortadaTwitter;
	}

	public void setImgPortadaTwitter(Image imgPortadaTwitter) {
		this.imgPortadaTwitter = imgPortadaTwitter;
	}

	public HorizontalLayout getHorizontalLayoutCentralImagen() {
		return horizontalLayoutCentralImagen;
	}

	public void setHorizontalLayoutCentralImagen(HorizontalLayout horizontalLayoutCentralImagen) {
		this.horizontalLayoutCentralImagen = horizontalLayoutCentralImagen;
	}

	public Element getVerticalLayoutBaneadoVerperfil() {
		return verticalLayoutBaneadoVerperfil;
	}

	public void setVerticalLayoutBaneadoVerperfil(Element verticalLayoutBaneadoVerperfil) {
		this.verticalLayoutBaneadoVerperfil = verticalLayoutBaneadoVerperfil;
	}

	public HorizontalLayout getHorizontalLayoutBaneados() {
		return horizontalLayoutBaneados;
	}

	public void setHorizontalLayoutBaneados(HorizontalLayout horizontalLayoutBaneados) {
		this.horizontalLayoutBaneados = horizontalLayoutBaneados;
	}

	public Button getButtonBaneados() {
		return buttonBaneados;
	}

	public void setButtonBaneados(Button buttonBaneados) {
		this.buttonBaneados = buttonBaneados;
	}

	public Image getImgPerfilUsuario() {
		return imgPerfilUsuario;
	}

	public void setImgPerfilUsuario(Image imgPerfilUsuario) {
		this.imgPerfilUsuario = imgPerfilUsuario;
	}

	public HorizontalLayout getHorizontalLayoutVerPerfilEscribirTweet() {
		return horizontalLayoutVerPerfilEscribirTweet;
	}

	public void setHorizontalLayoutVerPerfilEscribirTweet(HorizontalLayout horizontalLayoutVerPerfilEscribirTweet) {
		this.horizontalLayoutVerPerfilEscribirTweet = horizontalLayoutVerPerfilEscribirTweet;
	}

	public Element getVerticalLayoutVerPerfilEscribirTweet() {
		return verticalLayoutVerPerfilEscribirTweet;
	}

	public void setVerticalLayoutVerPerfilEscribirTweet(Element verticalLayoutVerPerfilEscribirTweet) {
		this.verticalLayoutVerPerfilEscribirTweet = verticalLayoutVerPerfilEscribirTweet;
	}

	public Button getButtonVerPerfil() {
		return buttonVerPerfil;
	}

	public void setButtonVerPerfil(Button buttonVerPerfil) {
		this.buttonVerPerfil = buttonVerPerfil;
	}

	public Button getButtonEscribirTweet() {
		return buttonEscribirTweet;
	}

	public void setButtonEscribirTweet(Button buttonEscribirTweet) {
		this.buttonEscribirTweet = buttonEscribirTweet;
	}

	public HorizontalLayout getHorizontalLayoutInferior() {
		return horizontalLayoutInferior;
	}

	public void setHorizontalLayoutInferior(HorizontalLayout horizontalLayoutInferior) {
		this.horizontalLayoutInferior = horizontalLayoutInferior;
	}

	public Element getVerticalLayoutInferior() {
		return verticalLayoutInferior;
	}

	public void setVerticalLayoutInferior(Element verticalLayoutInferior) {
		this.verticalLayoutInferior = verticalLayoutInferior;
	}

	public Element getVerticalLayoutDerecho() {
		return verticalLayoutDerecho;
	}

	public void setVerticalLayoutDerecho(Element verticalLayoutDerecho) {
		this.verticalLayoutDerecho = verticalLayoutDerecho;
	}

	public HorizontalLayout getHorizontalLayoutListaHashtagsUsados() {
		return horizontalLayoutListaHashtagsUsados;
	}

	public void setHorizontalLayoutListaHashtagsUsados(HorizontalLayout horizontalLayoutListaHashtagsUsados) {
		this.horizontalLayoutListaHashtagsUsados = horizontalLayoutListaHashtagsUsados;
	}

	public Element getVerticalLayoutListaHashtagsMasUsados() {
		return verticalLayoutListaHashtagsMasUsados;
	}

	public void setVerticalLayoutListaHashtagsMasUsados(Element verticalLayoutListaHashtagsMasUsados) {
		this.verticalLayoutListaHashtagsMasUsados = verticalLayoutListaHashtagsMasUsados;
	}

	public Div getLabelListaDeHashtagsMasUsados() {
		return labelListaDeHashtagsMasUsados;
	}

	public void setLabelListaDeHashtagsMasUsados(Div labelListaDeHashtagsMasUsados) {
		this.labelListaDeHashtagsMasUsados = labelListaDeHashtagsMasUsados;
	}

	public HorizontalLayout getHorizontalLayoutHashtagsMasUsados() {
		return horizontalLayoutHashtagsMasUsados;
	}

	public void setHorizontalLayoutHashtagsMasUsados(HorizontalLayout horizontalLayoutHashtagsMasUsados) {
		this.horizontalLayoutHashtagsMasUsados = horizontalLayoutHashtagsMasUsados;
	}

	public HorizontalLayout getHorizontalLayoutVerMasHashtags() {
		return horizontalLayoutVerMasHashtags;
	}

	public void setHorizontalLayoutVerMasHashtags(HorizontalLayout horizontalLayoutVerMasHashtags) {
		this.horizontalLayoutVerMasHashtags = horizontalLayoutVerMasHashtags;
	}

	public Element getVerticaLayoutVerMásHashTags() {
		return verticaLayoutVerMásHashTags;
	}

	public void setVerticaLayoutVerMásHashTags(Element verticaLayoutVerMásHashTags) {
		this.verticaLayoutVerMásHashTags = verticaLayoutVerMásHashTags;
	}

	public Button getButtonVerMasHashtags() {
		return buttonVerMasHashtags;
	}

	public void setButtonVerMasHashtags(Button buttonVerMasHashtags) {
		this.buttonVerMasHashtags = buttonVerMasHashtags;
	}



	
	

}